pkg/apis/observability-ui/v1alpha1/types.go

```diff
@@ -0,0 +1,120 @@
+// +groupName=observabilityui.rhobs
+// +kubebuilder:rbac:groups=observabilityui.rhobs,resources=observabilityuiplugins,verbs=list;get;watch
```

Comment:
Group names were recently discussed in the Obsservability group context. Seems like we're going to align on `observability.openshift.io` without any further subdomains. So Maybe we should go with `observabilityuiplugins.observability.openshift.io` here?
See also https://redhat-internal.slack.com/archives/C02BY4191B6/p1712866056544739

Reply:
(nit) uiplugins.observability.openshift.io to avoid repetition?